### PR TITLE
Bugfix: base quote estimated deposit different than user input

### PR DESCRIPTION
**Pull request by schystz**
**Important Note:
This branch base is from https://github.com/HaloDAO/halodao-interface/pull/94 so please consider checking that PR first before this issue so there are less files here to review.**

https://halodao.atlassian.net/browse/HDB-29

## Description of Changes

Fixed an issue wherein on XSGD/USDC curve wherein upon entering a base amount, the deposit amount returned by `viewDeposit` is different (much larger). This issue also causes the estimated LP amount to be too high so unless user sets a very high slippage (e.g. 50%) the tx will always fail.

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [ ] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDEV-XXX-description`
* [x] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by barryhalo — APPROVED** [review]
Looking good!

File: src/constants/errors.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export enum ZapErrorCode {
```
Comment (tracyarciaga) [inline src/constants/errors.ts:1]: @schystz @0xAplki Maybe we need to consolidate all our errors in one implementation, thoughts?

File: src/halo-hooks/amm/useAddRemoveLiquidity.ts [hunk 2]
```diff
@@ -61,50 +65,85 @@ export const useAddRemoveLiquidity = (address: string, token0: Token, token1: To
...
+        deposit: adjustedNumeraire,
+        lpToken,
+        base,
+        quote
+      }
+    },
+    [viewDeposit]
+  )
 
   const previewDepositGivenQuote = useCallback(
```
Comment (chris247474) [inline src/halo-hooks/amm/useAddRemoveLiquidity.ts:83]: @schystz i guess we wanna apply the new logic here as well to account for all scenarios

**Review by chris247474 — CHANGES_REQUESTED** [review]
@schystz works great on kovan and mainnet! one comment to implement the logic on `previewDepositGivenQuote` as well

File: src/halo-hooks/amm/useAddRemoveLiquidity.ts (hunk 2 above)
Comment (schystz) [reply src/halo-hooks/amm/useAddRemoveLiquidity.ts:83]: Already implemented here @ctverceles , pls see lines 91-108.

I think you might have seen the outdated commit I posted earlier. Added a few more commits after that.

**Comment by chris247474** [comment]
Looks good @schystz ! tried on kovan, polygon mainnet pools. estimation in line err shows up as expected when new estimate is very slightly over user input. Re requested review but this is approval

**Review by tracyarciaga — APPROVED** [review]
lgtm

File: src/constants/errors.ts (hunk 1 above)
Comment (schystz) [reply src/constants/errors.ts:1]: Definitely! I think @0xAplki is working on that already right?

Though to clarify, this file is meant to be a collection of errors returns by web3/contracts though, not the errors we show to the user.

**Pull request merged**